ethcore/evm/src/interpreter/gasometer.rs

```diff
@@ -124,13 +124,18 @@ impl<Gas: evm::CostType> Gasometer<Gas> {
 				let address = H256::from(stack.peek(0));
 				let newval = stack.peek(1);
 				let val = U256::from(&*ext.storage_at(&address)?);
+				let orig = U256::from(&*ext.initial_storage_at(&address)?);
```

Comment:
I still believe it's better to move it inside the if banch

Reply:
Sorry I missed this. Moved!